EIPS/eip-5659.md

```diff
@@ -0,0 +1,47 @@
+---
+eip: 5659
+title: Social Media URI Propagation Event
+description: A minimal primitive to emit URI events for social media.
+author: Tim Daubenschütz (@TimDaub), Auryn Macmillan (@auryn-macmillan)
+discussions-to: https://ethereum-magicians.org/t/social-media-uri-propagation-event/10893
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-09-15
+---
+
+## Abstract
+
+We introduce a minimal on-chain primitive to emit URIs using Ethereum's event log infrastructure.
+
+## Motivation
+
+The modern social network's base layer is distributing content behind Universal Resource Identifiers as defined by RFC 3986. Any web resource can be identified by an URI or a URL and so upgrading Ethereum with a standard way of distributing content is useful.
+
+## Specification
+
+```solidity
+contract Propagator {
+  event NewPost(address indexed author, string indexed uri);
+  function submit(string calldata uri) external {
+    emit NewPost(msg.sender, uri);
+  }
```

Comment:
Is this implementation the only valid implementation? If not, you should probably switch this to an interface and let implementers decide how to implement it.